daemon/src/position_metrics.rs

```diff
@@ -0,0 +1,344 @@
+use crate::db;
+use async_trait::async_trait;
+use futures::StreamExt;
+use model::CfdEvent;
+use model::EventKind;
+use model::OrderId;
+use model::Position;
+use model::Usd;
+use rust_decimal::prelude::FromPrimitive;
+use rust_decimal::Decimal;
+use std::time::Duration;
+use tokio_tasks::Tasks;
+use xtra_productivity::xtra_productivity;
+use xtras::SendInterval;
+
+// TODO: ideally this would be more often
+pub const UPDATE_METRIC_INTERVAL_SECONDS: Duration = Duration::from_secs(60);
+
+pub struct Actor {
+    tasks: Tasks,
+    db: db::Connection,
+}
+
+impl Actor {
+    pub fn new(db: db::Connection) -> Self {
+        Self {
+            db,
+            tasks: Tasks::default(),
+        }
+    }
+}
+
+#[async_trait]
+impl xtra::Actor for Actor {
+    type Stop = ();
+
+    async fn started(&mut self, ctx: &mut xtra::Context<Self>) {
+        let this = ctx.address().expect("we are alive");
+        self.tasks
+            .add(this.send_interval(UPDATE_METRIC_INTERVAL_SECONDS, || UpdateMetrics));
+    }
+
+    async fn stopped(self) -> Self::Stop {}
+}
+
+#[xtra_productivity]
+impl Actor {
+    async fn handle(&mut self, _: UpdateMetrics) {
+        tracing::debug!("Collecting metrics");
+        let mut stream = self.db.load_all_cfds::<Cfd>(());
+
+        let mut cfds = Vec::new();
+        while let Some(cfd) = stream.next().await {
+            let cfd = match cfd {
+                Ok(cfd) => cfd,
+                Err(e) => {
+                    tracing::error!("Failed to rehydrate CFD: {e:#}");
+                    continue;
+                }
+            };
+            cfds.push(cfd);
+        }
```

Comment:
I suspect this would be more idiomatic (and faster):

```suggestion
              let stream = self.db.load_all_cfds::<Cfd>(());

        let cfds: Vec<Cfd> = stream
            .inspect(|cfd| {
                if let Err(e) = cfd {
                    tracing::error!("Failed to rehydrate CFD: {e:#}");
                }
            })
            .filter_map(|cfd| async move { cfd.ok() })
            .collect()
            .await;

```